### PR TITLE
gh-101561: Add typing.override decorator

### PR DESCRIPTION
**Pull request by stroxler**
The code is pulled almost unchanged from `typing_extensions`:

- [decorator implementation](https://github.com/python/typing_extensions/blob/main/src/test_typing_extensions.py#L168-L202)
- [tests](https://github.com/python/typing_extensions/blob/main/src/test_typing_extensions.py#L168-L202)

Question that came up was where to put the type variable and what to name it:
- In `typing_extensions` the type variables use underscore prefixes to indicate that they are not public, but `typing` doesn't appear to use that convention so I named the type variable `F`.
- I couldn't put the type variable next to all of the other ones (used in generic container definitions) because it relies on `Callable` which is defined below. If this is a problem I could correct it using a forward reference, or we could try to rearrange the module to put special forms above type variables.

## Testing the code:

First set up the repo, following instructions
at https://devguide.python.org/ by running:
```
./configure --with-pydebug && make -j
```

Then run the typing tests:
```
./python -m test test_typing -v
```

I ran the full test suite with
```
./python -m test -j3
```
and it came back clean except for a `test_grp` failure which I seem to get on trunk as well - likely something in my build is misconfigured but I'm pretty sure it is unrelated to the changes here.

* Issue: gh-101561

## Make sure the docs build:

```
cd Doc
make venv
make html
```

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Thanks! Haven't looked in depth yet at all, but we'll also need:
- A news entry, as @bedevere-bot says
- An entry in `Doc/whatsnew/3.12.rst`
- (optionally) you can add your name to `Misc/ACKS`

File: Doc/library/typing.rst
Comment (JelleZijlstra) [inline Doc/library/typing.rst]:
```suggestion
   ``if getattr(obj, "__override__", False)`` can be used at runtime to determine
   whether an object ``obj`` has been marked as an override.  If the decorated object
```

Comment (stroxler) [reply Doc/library/typing.rst]: Yikes, you can tell I copied this to get the formatting right.

File: Misc/NEWS.d/next/Library/2023-02-04-16-35-46.gh-issue-101561.Xo6pIZ.rst
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-02-04-16-35-46.gh-issue-101561.Xo6pIZ.rst]:
```suggestion
See :pep:`698` for details. Patch by Steven Troxler.
```

File: Lib/typing.py
Comment (TeamSpen210) [inline Lib/typing.py]: `typing_extensions` has to use this workaround since it supports Python 3.7, but `typing` can just use a positional-only arg. Might also be better to name it something more meaningful like `func` or `method`?

**Comment by Fidget-Spinner** [comment]
The implementation for override LGTM.

**Comment by stroxler** [comment]
(Rebase on main)

@gvanrossum I *think* this is waiting on review from you

File: Doc/whatsnew/3.12.rst
Comment (JelleZijlstra) [inline Doc/whatsnew/3.12.rst]:
```suggestion
  same name on a base class, as per :pep:`698`. (Contributed by Steven Troxler in
  :gh:`101564`.)
```
punctuation nit

File: Misc/NEWS.d/next/Library/2023-02-04-16-35-46.gh-issue-101561.Xo6pIZ.rst
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2023-02-04-16-35-46.gh-issue-101561.Xo6pIZ.rst]:
```suggestion
Add a new decorator :func:`typing.override`. See :pep:`698` for details. Patch by Steven Troxler.
```
This can be kept more concise. The What's New can have more of an explanation.

**Review by JelleZijlstra — APPROVED** [review]
Sorry for dropping this; I have a few nits but I'm happy to merge the PR. I'll leave it open for a few days in case Guido does want to take a look himself.

**Comment by gvanrossum** [comment]
I defer to @JelleZijlstra if that's okay.

**Comment by sobolevn** [comment]
Sorry for being late with a review, I've opened a follow-up for this: https://github.com/python/cpython/issues/102324 👍 

Thanks a lot for adding this decorator! Great work, it would be very useful! 👍

**Pull request merged**
